test/unit/sqs/index.test.ts

```diff
@@ -22,140 +23,296 @@ describe("SqsQueue tests", () => {
 		handle: jest.fn()
 	};
 
+	const mockErrorHandler = {
+		handle: jest.fn()
+	};
+
 	const generatePayload = (): TestMessage => ({ msg: uuidv4() });
 
-	const createSqsQueue = () => {
+	const createSqsQueue = (timeout: number) => {
 		return new SqsQueue({
 			queueName: TEST_QUEUE_NAME,
 			queueUrl: TEST_QUEUE_URL,
 			queueRegion: TEST_QUEUE_REGION,
-			longPollingIntervalSec: 0
+			longPollingIntervalSec: 0,
+			timeoutSec: timeout,
+			maxAttempts: 3
 		},
-		mockRequestHandler);
+		mockRequestHandler,
+		mockErrorHandler);
 	};
 
 	let queue: SqsQueue<TestMessage>;
 
-	beforeEach(() => {
-		queue = createSqsQueue();
-		queue.start();
-	});
+	describe("Normal execution tests", () => {
 
-	afterEach(async () => {
-		queue.stop();
-		await queue.waitUntilListenerStopped();
-	});
+		beforeEach(() => {
+			queue = createSqsQueue(10);
+			queue.start();
+		});
 
-	test("Message gets received", (done: DoneCallback) => {
+		afterEach(async () => {
+			queue.stop();
+			await queue.waitUntilListenerStopped();
+		});
+
+		test("Message gets received", (done: DoneCallback) => {
 
-		const testPayload = generatePayload();
+			const testPayload = generatePayload();
 
-		mockRequestHandler.handle.mockImplementation((context: Context<TestMessage>) => {
-			expect(context.payload).toStrictEqual(testPayload);
-			done();
+			mockRequestHandler.handle.mockImplementation((context: Context<TestMessage>) => {
+				expect(context.payload).toStrictEqual(testPayload);
+				done();
+			});
+			queue.sendMessage(testPayload);
 		});
-		queue.sendMessage(testPayload);
-	});
 
 
-	test("Queue is restartable", async (done: DoneCallback) => {
+		test("Queue is restartable", async (done: DoneCallback) => {
 
-		const testPayload = generatePayload();
+			const testPayload = generatePayload();
 
-		mockRequestHandler.handle.mockImplementation((context: Context<TestMessage>) => {
-			expect(context.payload).toStrictEqual(testPayload);
-			done();
+			mockRequestHandler.handle.mockImplementation((context: Context<TestMessage>) => {
+				expect(context.payload).toStrictEqual(testPayload);
+				done();
+			});
+
+			queue.stop();
+
+			//delaying to make sure all asynchronous invocations inside the queue will be finished and it will stop
+			await queue.waitUntilListenerStopped();
+
+			queue.start();
+
+			queue.sendMessage(testPayload);
 		});
 
-		queue.stop();
+		test("Message received with delay", (done: DoneCallback) => {
 
-		//delaying to make sure all asynchronous invocations inside the queue will be finished and it will stop
-		await queue.waitUntilListenerStopped();
+			const testPayload = generatePayload();
+			const receivedTime = {time: new Date().getTime()};
 
-		queue.start();
+			mockRequestHandler.handle.mockImplementation((context: Context<TestMessage>) => {
 
-		queue.sendMessage(testPayload);
-	});
+				if(context.payload.msg !== testPayload.msg) {
+					//discard message if it is not our test message
+					return;
+				}
```

Comment:
What's this?

Reply:
As the comment says: Discard message if it is not our test message.

To make sure if there are some messages from other tests, for any reason, pollute the queue, the test won't be affected.

Reply:
What I'm asking is _why is it in your test and why is your state being polluted between tests?_

This is not how tests are suppose to be done and is extremely risky for getting flaky tests or false positives.